packages/openchs-android/src/views/filter/FiltersView.js

```diff
@@ -144,6 +153,31 @@ class FilterView extends AbstractComponent {
         this.dispatchAction(FilterActionNames.ADD_PROGRAM, {programUUID: uuid});
     }
 
+    clearData() {
+        this.dispatchAction(FilterActionNames.ON_DATE_CLEAR,
+            {filterDate: this.state.filterDate})
+        for(len in this.state.selectedGeneralEncounterTypes){
+            this.dispatchAction(FilterActionNames.ADD_GENERAL_VISITS,
+                {encounterUUID: this.state.selectedGeneralEncounterTypes[len]['uuid']})}
+        var MaxAttempts = 3;
+        while (MaxAttempts>=0){
```

Comment:
what is the maxAttempts? If this is used for address hierarchy we should recursively fetch that because it'll fail for hierarchy more than 3.